src/react/hooks/use-room.ts

```diff
@@ -0,0 +1,126 @@
+import { ConnectionStatusChange, Room, RoomLifecycle, RoomStatusChange } from '@ably/chat';
+import * as Ably from 'ably';
+import { useCallback, useContext, useEffect, useState } from 'react';
+
+import { ChatStatusResponse } from '../chat-status-response.js';
+import { RoomContext } from '../contexts/room-context.js';
+import { useChatConnection } from './use-chat-connection.js';
+
+/**
+ * Parameters for the useRoom hook.
+ */
+export interface UseRoomParams {
+  /**
+   * Callback for when the room status changes. The listener is removed when
+   * the component unmounts.
+   * @param change The change object.
+   */
+  onStatusChange?: (change: RoomStatusChange) => void;
+
+  /**
+   * Callback for when the connection status changes. The listener is removed
+   * when the component unmounts.
+   * @param change The change object.
+   */
+  onConnectionStatusChange?: (change: ConnectionStatusChange) => void;
+}
+
+/**
+ * The return type for the useRoom hook.
+ */
+export interface UseRoomResponse extends ChatStatusResponse {
+  /** The room object. */
+  room: Room;
+
+  /**
+   * Shortcut to room.attach. Not needed if room provider has attach=true set,
+   * which is the default behavior.
+   */
+  attach: () => Promise<void>;
+
+  /**
+   * Shortcut to room.detach. Not needed if room provider has attach=true
+   * set, which is the default behavior.
+   */
+  detach: () => Promise<void>;
+}
+
+// Helper function to create a status object from a a change event or the room
```

Comment:
Could we convert this to a multi-line comment since it describes a function, also there were some minor grammar changes :) 

/**
Helper function to create a status object from a change event or the room
status. It makes sure `error` isn't set at all if it's undefined in the source.
*/